src/VirtualClient/VirtualClient.Monitors/AmdSmiMonitor.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Monitors
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO.Abstractions;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using global::VirtualClient;
+    using global::VirtualClient.Contracts;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
+    using VirtualClient.Common;
+    using VirtualClient.Common.Extensions;
+    using VirtualClient.Common.Telemetry;
+
+    /// <summary>
+    /// The Performance Counter Monitor for Virtual Client
+    /// </summary>
+    public class AmdSmiMonitor : VirtualClientIntervalBasedMonitor
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AmdSmiMonitor"/> class.
+        /// </summary>
+        public AmdSmiMonitor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+            : base(dependencies, parameters)
+        {
+        }
+
+        /// <inheritdoc/>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            switch (this.Platform)
+            {
+                case PlatformID.Win32NT:
+                    await this.QueryGpuAsync(telemetryContext, cancellationToken)
+                        .ConfigureAwait(false);
+                    break;
+
+                case PlatformID.Unix:
+                    // not supported at the moment
+                    break;
+            }
+        }
+
+        /// <inheritdoc/>
+        protected void ValidateParameters()
+        {
+            if (this.MonitorFrequency <= TimeSpan.Zero)
+            {
+                throw new MonitorException(
+                    $"The monitor frequency defined/provided for the '{this.TypeName}' component '{this.MonitorFrequency}' is not valid. " +
+                    $"The frequency must be greater than zero.",
+                    ErrorReason.InvalidProfileDefinition);
+            }
+        }
+
+        /// <summary>
+        /// Query the gpu for utilization information
+        /// </summary>
+        /// <param name="telemetryContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        private async Task QueryGpuAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            ISystemManagement systemManagement = this.Dependencies.GetService<ISystemManagement>();
+            IFileSystem fileSystem = systemManagement.FileSystem;
+
+            int totalSamples = (int)this.MonitorFrequency.TotalSeconds;
+            string command = "amdsmi";
+            string commandArguments = "metric --csv";

```

Comment:
Just out of precaution, is there a way to deterministically pass in the columns of the csv output? So that the order of the metrics won't change?